engine/src/main/java/org/terasology/rendering/world/WorldRenderer.java

```diff
@@ -23,60 +23,227 @@
+     *
+     * In mono mode, used to render to a standard monitor, the rendering engine goes through
+     * a single rendering stage every frame, called MONO.
+     *
+     * Stereo mode is for stereoscopic displays: every frame the rendering engine first go through
+     * the LEFT_EYE rendering stage and then the RIGHT_EYE rendering stage. Each stage produces
+     * an image and the two are then combined at the end of the RIGHT_EYE stage.
+     *
+     * Notice that the renderer has no explicit notion of mono/stereo. It only knows which stage is the current one.
+     */
+    // TODO: candidate for renaming simply to RenderingStage
+    enum WorldRenderingStage {
         MONO,
         LEFT_EYE,
         RIGHT_EYE
     }
 
+    /**
+     * Retrieves the camera the world is being rendered from.
+     *
+     * @return a Camera object
+     */
     Camera getActiveCamera();
 
+    /**
+     * Retrieves the camera positioned and oriented to look down on the world from the point of view
+     * of the main light source. This camera is used to produce the shadow map.
+     *
+     * @return a Camera object
+     */
     Camera getLightCamera();
 
+    /**
+     * Retrieves the chunk provider currently in use by the renderer.
+     *
+     * @return a ChunkProvider object.
+     */
+    // TODO: candidate for removal - can be obtained through a context object.
     ChunkProvider getChunkProvider();
 
+    // TODO: candidate for removal - nothing in the engine seems to use it
     void setPlayer(LocalPlayer localPlayer);
 
+    /**
+     * Called potentially multiple times per frame, this method triggers the update of any data structure
+     * the renderer might need. The rendering itself happens only in the "render" method.
+     *
+     * @param delta The elapsed time, in seconds, since the previous update.
+     */
     void update(float delta);
 
+    /**
+     * This method triggers the execution of the rendering pipeline and, eventually, sends the output to the display
+     * or to a file, when grabbing a screenshot.
+     *
+     * @param renderingStage "MONO" for standard rendering and "LEFT_EYE" or "RIGHT_EYE" for stereoscopic displays.
+     */
     void render(WorldRenderingStage renderingStage);
 
+    /**
+     * Gives the chance to an implementation to dispose any internal reference that might need a more careful disposal.
+     */
     void dispose();
 
+    /**
+     * Used only during the early part of the WorldRenderer lifecycle, this method triggers the loading of
+     * chunks around the camera and the generation of their meshes.
+     *
+     * @return Returns True if all necessary chunks have been loaded and their meshes have been generated, False otherwise.
+     */
+    // TODO: it might be desirable to separate the loading of chunks and mesh generation. The former may concern
+    // TODO: systems dealing with beyond-the-horizon simulations. The latter usually only concerns the renderer.
```

Comment:
These TODOs are more a long term discussion/goal. The basic thinking here is that the renderer shouldn't really be concerned with loading/unloading chunks, but only with generating and unloading their meshes. By splitting these two concerns the volume of loaded chunks could be larger than what is visible and entities and simulations could be enabled even if beyond the horizon. For example, a wild fire might initiate beyond the visible range and only eventually, as it spreads, become visible. Or a volcano might erupt beyond the visible range but lava flows might eventually reach the player.